libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -1056,3 +1063,24 @@ float AC_WPNav::get_slow_down_speed(float dist_from_dest_cm, float accel_cmss)
         return target_speed;
     }
 }
+
+/// wp_speed_update - calculates how to handle speed change requests
+void AC_WPNav::wp_speed_update(float dt)
+{
+    // calculate speed change for steady-state or speeding up
+    if (_wp_desired_speed_cms >= _wp_speed_cms) {
+        _wp_speed_cms = _wp_desired_speed_cms;
```

Comment:
Is there any reason we don't slew the speed up and down?

Reply:
I tried that and it didn't seem to make any difference. Feed the PSC a new speed and it somehow seems to handle that by itself. I tested with a Ch6 tuning knob set from 5 to 20 m/s range with both my 1.2m TriCopter and helicopter. It accelerates very smoothly and gradual when the knob is turned to max. And I don't understand why.  I originally had it coded to use WPNAV_ACCEL on speedup request but I removed it after testing it because it took forever to get it up to the requested speed. Again, I don't understand why. It was only the slowdown that had a problem with sudden request for lower speed.

Reply:
My theory is that the reason it accelerates smoothly without slewing it is because it is buffered by the re-calculation of the leash length. The leash stretches like a rubber band as speed increases. This provides a natural buffer as the target gets further away from the actual aircraft. The inverse is true on slow down request.